src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java

```diff
@@ -374,6 +374,17 @@ public static String getPathWithoutBucket(String path) {
 
     }
 
+    /**
+     * String -> Path. This *should* not be necessary (use Paths.get(URI.create(...)) instead) , but it currently is
+     * on Spark because using the fat, shaded jar breaks the registration of the GCS FilesystemProvider.
```

Comment:
[The FileSystems API page](https://docs.oracle.com/javase/7/docs/api/java/nio/file/FileSystems.html) describes the mechanism. If our jar file is not in the classpath but is instead loaded dynamically, this could explain the filesystem not being properly installed.